hedera-node/src/main/java/com/hedera/services/ledger/properties/AccountProperty.java

```diff
@@ -78,10 +78,13 @@
 		public BiConsumer<MerkleAccount, Object> setter() {
 			return (a, v) -> {
 				try {
-					a.setBalance((long) v);
+					a.setBalance(((Number)v).longValue());
+				} catch (ClassCastException cce) {
+					throw new IllegalArgumentException(
+							"Wrong argument type. Argument needs to be of type int or long :" + v.toString());
```

Comment:
`"Wrong argument type! Argument needs to be of type int or long. Actual value: " + v.toString()`
and also add `cce` to be wrapped in this `IllegalArgumentException`

Reply:
Is there special reason to wrap the `cce`? I think the new description is clear enough.

Reply:
For debugging purpose, it is always better to have all the info. If you don't wrap the cce, you will lose some info there. Another way to handle this is to log the cce.

Reply:
I see and added. I had the `cce` originally and notices that the message was fairly clear to me. But if other developers look at it, it may not be obvious where the error comes from.